lib/publiccloud/utils.pm

```diff
@@ -193,9 +193,7 @@ sub is_hardened() {
 sub is_embargo_update {
     my ($incident, $type) = @_;
     return 0 if ($type =~ /PTF/);
-    script_retry("curl -sSf https://build.suse.de/attribs/SUSE:Maintenance:$incident -o /tmp/$incident.txt");
-    return 1 if (script_run("grep 'OBS:EmbargoDate' /tmp/$incident.txt") == 0);
-    return 0;
+    return script_output("curl -sSf https://smelt.suse.de/api/v1/basic/incidents/$incident/ | jq .embargo") ne 'null';
```

Comment:
Issue before was that connection issue to the server result in function to return False.
Is it different now?

I'm thinking to 3 behaviors:
- connection to smelt is fine, returned text is a valid JSON, JSON has embargo  --> return True
- connection to smelt is fine, returned text is a valid JSON, JSON has NO embargo  --> return False
- anything else `croak` or `die`

Reply:
+1 on considering the case where SMELT is unreachable

Reply:
are you not missing 3 `die` or `croak` ?

Reply:
WDYT about : 

`my $cmd = "set -o pipefail; curl -sSf https://smelt.suse.de/api/v1/basic/incidents/$incident/ | jq .embargo; set +o pipefail";`

this way any issues while querying smelt or processing JSON will made corresponding `script_output` call to die . I assume this will cover your expectations ?

Reply:
yes